Fix type errors on other models' managers when using objects = models.Manager() in Model.

syastrov opened the pull request:
I was getting all sorts of strange type errors on other models just because I had `objects` defined on one of my models as models.Manager():
```
class Foo(models.Model):
    objects = models.Manager()

class Bar(models.Model):
    a = models.IntegerField()

Bar.objects.get(id=1).a # Type error saying no attribute 'a' is defined on 'Foo'.
```

I made the following changes:
- Fill in the type argument for that particular instance of the manager, rather than modifying the bases of the Manager type.
- Instantiate a new Instance in `determine_proper_manager_type` so the code doesn't crash under mypy-mypyc.

syastrov committed:
Fix bug where models with a class variable using a manager defined would interfere with other managers.

- Fill in the type argument for that particular instance of the manager, rather than modifying the bases of the Manager type.
- Instantiate a new Instance from determine_proper_manager_type so The code doesn't crash under mypy-mypyc.

mkurnikov commented on mypy_django_plugin/main.py:
```diff
@@ -55,13 +55,20 @@ def determine_proper_manager_type(ctx: FunctionContext) -> Type:
+    if has_manager_base:
+        # Fill in the manager's type argument from the outer model
+        new_type_args = [Instance(outer_model_info, [])]
+        return Instance(typ=ret.type, args=new_type_args, line=ret.line, column=ret.column)
```
`helpers.reparametrize_instance` could be used here.

syastrov replied:
👍

syastrov committed:
Merge branch 'master' into bugfix-managers-on-unrelated-models-influencing-each-other

syastrov committed:
Use helpers.reparametrize_instance per review comment.

syastrov committed:
Merge branch 'master' into bugfix-managers-on-unrelated-models-influencing-each-other

syastrov committed:
Updated ignored errors in Django test for get_objects_or_404.

- For some reason, `Manager[nothing]` is now removed from expected types.
  However, I think this makes sense anyway, as Manager is a subclass of QuerySet.
